cluster/kubernetes/resourcekinds.go

```diff
@@ -263,21 +263,40 @@ func makeStatefulSetPodController(statefulSet *apiapps.StatefulSet) podControlle
 	var status string
 	objectMeta, statefulSetStatus := statefulSet.ObjectMeta, statefulSet.Status
 
-	status = StatusUpdating
+	status = StatusStarted
 	rollout := cluster.RolloutStatus{
-		Desired:  *statefulSet.Spec.Replicas,
-		Updated:  statefulSetStatus.UpdatedReplicas,
-		Ready:    statefulSetStatus.ReadyReplicas,
-		Outdated: statefulSetStatus.CurrentReplicas - statefulSetStatus.UpdatedReplicas,
-		// TODO Add Messages after "ODO: Add valid condition types for Statefulsets." fixed in
+		Desired: *statefulSet.Spec.Replicas,
+		Ready:   statefulSetStatus.ReadyReplicas,
+		// There is no Avaliable parameter for statefulset, so use Ready instead
+		Available: statefulSetStatus.ReadyReplicas,
+		// TODO Add Messages after "TODO: Add valid condition types for Statefulsets." fixed in
 		// https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/pkg/apis/apps/types.go#L205
 	}
 
+	// rolling update
+	if statefulSet.Spec.UpdateStrategy.Type == apiapps.RollingUpdateStatefulSetStrategyType && statefulSet.Spec.UpdateStrategy.RollingUpdate != nil {
+		if statefulSet.Spec.Replicas != nil && statefulSet.Spec.UpdateStrategy.RollingUpdate.Partition != nil {
+			// Desired for this partition
```

Comment:
for the uneducated (like myself), would you mind explaining somewhere why we subtract the partition here?

Edit: looked it up and I understand now. including the link wouldn't hurt i guess https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#partitions